app/services/xccdf_report_parser.rb

```diff
@@ -54,28 +63,46 @@ def save_host
     new_profiles = host_new_profiles
     @host.profiles << new_profiles if new_profiles.present?
     inventory_api.sync
+    invalidate_cache
+    @host.reload
+  end
+
+  def host_profiles
+    test_result_profile = @profiles.find do |profile|
+      profile.ref_id == @test_result_file.test_result.profile_id
+    end
+
+    unless @host.profiles.pluck(:ref_id).include?(test_result_profile.ref_id)
+      host_profile = test_result_profile.dup
+      host_profile.update(account: @account, rules: test_result_profile.rules)
+
+      return @host.profiles | [host_profile]
```

Comment:
Not sure what exactly do you mean to return early here?

Reply:
This returns the union of the two arrays. `(The existing host profiles) UNION ([One more profile, maybe new, maybe already existing])`